static/js/components/email/lib.js

```diff
@@ -25,28 +31,65 @@ export const COURSE_TEAM_EMAIL_CONFIG: EmailConfig = {
     subheading: `${course.title} Course Team`
   }),
 
-  emailSendAction: (emailState) => (
-    sendCourseTeamMail(
-      emailState.inputs.subject || '',
-      emailState.inputs.body || '',
-      emailState.params.courseId
-    )
-  )
+  getEmailSendFunction: () => sendCourseTeamMail,
+
+  emailSendParams: (emailState) => ([
+    emailState.inputs.subject || '',
+    emailState.inputs.body || '',
+    emailState.params.courseId
+  ])
 };
 
 export const SEARCH_RESULT_EMAIL_CONFIG: EmailConfig = {
   title: 'New Email',
 
   emailOpenParams: (searchkit: Object) => ({
-    params: {},
     subheading: `${searchkit.getHitsCount() || 0} recipients selected`
   }),
 
-  emailSendAction: (emailState) => (
-    sendSearchResultMail(
-      emailState.inputs.subject || '',
-      emailState.inputs.body || '',
-      emailState.searchkit.buildQuery().query
-    )
-  )
+  getEmailSendFunction: () => sendSearchResultMail,
+
+  emailSendParams: (emailState) => ([
+    emailState.inputs.subject || '',
+    emailState.inputs.body || '',
+    emailState.searchkit.buildQuery().query
+  ])
+};
+
+export const LEARNER_EMAIL_CONFIG: EmailConfig = {
```

Comment:
Can you add tests which cover these functions, either via unit tests and mocks or with an integration test?